tests/verify_aaefold_rangerepair.erl

```diff
@@ -0,0 +1,253 @@
+          ]},
+         {riak_core,
+          [
+           {ring_creation_size, ?DEFAULT_RING_SIZE}
+          ]}]
+       ).
+-define(NUM_NODES, 4).
+-define(NUM_KEYS, 20000).
+-define(BUCKET, <<"test_bucket">>).
+-define(N_VAL, 3).
+-define(DELTA_COUNT, 10).
+
+confirm() ->
+    lager:info("Testing without rebuilds - using http api"),
+    Nodes0 = rt:build_cluster(?NUM_NODES, ?CFG_NOREBUILD),
+    CH_HTTP = rt:httpc(hd(Nodes0)),
+    ok = verify_aae_repair(Nodes0, rhc, CH_HTTP, Nodes0),
+
+    rt:clean_cluster(Nodes0),
+
+    Nodes1 = rt:build_cluster(?NUM_NODES, ?CFG_NOREBUILD),
+    CH_PB = rt:pbc(hd(Nodes1)),
+    ok = verify_aae_repair(Nodes1, riakc_pb_socket, CH_PB, Nodes1),
+
+    pass.
+
+
+verify_aae_repair(Nodes, ClientMod, ClientHead, Nodes) ->
+
+    TailNode = lists:last(Nodes),
+    HeadNode = hd(Nodes),
+
+    lager:info("Generating KVs (3 lots)"),
+
+    TestKVs0 = test_data(1, ?NUM_KEYS, list_to_binary("U1")),
+    TestKVs1 = test_data(?NUM_KEYS + 1, ?NUM_KEYS * 2, list_to_binary("U2")),
+    TestKVs2 = test_data(?NUM_KEYS * 2 + 1, ?NUM_KEYS * 3, list_to_binary("U3")),
+
+    TS0 = os:timestamp(),
+    timer:sleep(1000),
+
+    lager:info("Commencing object load"),
+    ok = write_data(HeadNode, TestKVs0),
+    lager:info("Loaded ~w objects", [length(TestKVs0)]),
+    wait_until_root_stable(ClientMod, ClientHead),
+
+    timer:sleep(1000),
+    TS1 = os:timestamp(),
+    timer:sleep(1000),
+
+    lager:info("Second object load"),
+    ok = write_data(HeadNode, TestKVs1),
+    lager:info("Loaded ~w objects", [length(TestKVs1)]),
+    wait_until_root_stable(ClientMod, ClientHead),
+
+    timer:sleep(1000),
+    TS2 = os:timestamp(),
+    timer:sleep(1000),
+
+    rt:stop_and_wait(TailNode),
+
+    lager:info("Third object load - with node down"),
+    ok = write_data(HeadNode, TestKVs2),
+    lager:info("Loaded ~w objects", [length(TestKVs2)]),
+    wait_until_root_stable(ClientMod, ClientHead),
+
+    lager:info("Cleaning data directory on stopped node"),
+    TestMetaData = riak_test_runner:metadata(),
+    KVBackend = proplists:get_value(backend, TestMetaData),
+    rt:clean_data_dir([TailNode], base_dir_for_backend(KVBackend)),
+
+    rt:start_and_wait(TailNode),
+
+    rt:wait_until_transfers_complete(Nodes),
+    
+    object_count(ClientMod, ClientHead, "after restart - particpating"),
+    object_count(ClientMod, ClientHead, "after restart - particpating"),
+    object_count(ClientMod, ClientHead, "after restart - particpating"),
+    object_count(ClientMod, ClientHead, "after restart - particpating"),
+    object_count(ClientMod, ClientHead, "after restart - particpating"),
+
+    rpc:call(TailNode, riak_client, remove_node_from_coverage, []),
+
+    TC1 = object_count(ClientMod, ClientHead, "afetr restart - not particpating"),
+    ?assertMatch(TC1, 3 * ?NUM_KEYS),
+
+    RR0 = get_read_repair_total(Nodes),
+
+    verify_tictac_aae:verify_data(HeadNode, TestKVs2),
+
+    lager:info("Forcing read repair - first set"),
```

Comment:
Add some more intuition as comments... this is simulating "reading back from backup", right?